xs2a-adapter-service-remote/src/test/java/de/adorsys/xs2a/tpp/ErrorResponseDecoder.java

```diff
@@ -0,0 +1,38 @@
+package de.adorsys.xs2a.tpp;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import de.adorsys.xs2a.adapter.service.ErrorResponse;
+import de.adorsys.xs2a.adapter.service.ResponseHeaders;
+import de.adorsys.xs2a.adapter.service.exception.ErrorResponseException;
+import feign.Response;
+import feign.codec.ErrorDecoder;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Configuration;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+@Configuration
+public class ErrorResponseDecoder implements ErrorDecoder {
+
+    @Autowired
+    ObjectMapper objectMapper;
+
+    @Override
+    public Exception decode(String methodKey, Response response) {
+        ResponseHeaders responseHeaders = ResponseHeaders.fromMap(
+            response.headers().entrySet().stream()
+                .collect(Collectors.toMap(Map.Entry::getKey, e -> e.getValue().iterator().next()))
+        );
+
+        ErrorResponse errorResponse = null;
+        try {
+            errorResponse = objectMapper.readValue(response.body().toString(), ErrorResponse.class);
```

Comment:
response.body().toString() is throwing JsonParseException
should be response.body().asInputStream()

Reply:
Could you provide a reproducible scenario?

Reply:
response.body() return value is of type (interface) feign.Body. This class defines methods asInputStream() and asReader() for extracting the response. The implemantation class feign.InputStreamBody for example does not override the toString() method.

See also
https://www.codota.com/code/java/methods/feign.Response/body

Reply:
I see your point.
https://github.com/adorsys/xs2a-adapter/pull/177